ui-tests/src/test/java/org/openmrs/reference/page/ManageLocationsPage.java

```diff
@@ -35,6 +48,67 @@ public AddEditLocationPage goToAddLocation() {
         return new AddEditLocationPage(this);
     }
 
+    public void goToAddNewLocation() {
+        clickOn(ADD_NEW_LOCATION_BUTTON);
+    }
+
+    public void goToEditLocationForm() {
+    	clickOn(EDIT_LOCATION_LINK);
+    }
+    
+    public void enterLocationName(String name) {     
+        findElement(LOCATION_NAME_FIELD).clear();
+        findElement(LOCATION_NAME_FIELD).sendKeys(name);
+    }
+
+    public void enterLocationDescription(String description) {
+        findElement(LOCATION_DESCRIPTION_FIELD).clear();
+        findElement(LOCATION_DESCRIPTION_FIELD).sendKeys(description);
+    }
+
+    public void enterAddress1(String address) {
+        findElement(LOCATION_ADDRESS_FIELD_1).clear();
+        findElement(LOCATION_ADDRESS_FIELD_1).sendKeys(address);
+    }
+    
+    public void enterAddress2(String address) {
+        findElement(LOCATION_ADDRESS_FIELD_2).clear();
+        findElement(LOCATION_ADDRESS_FIELD_2).sendKeys(address);
+    }
+    
+    public void enterState(String stateName) {
+        findElement(STATE_FIELD).clear();
+        findElement(STATE_FIELD).sendKeys(stateName);
+    }
+
+    public void enterCity(String cityName) {
+        findElement(CITY_FIELD).clear();
+        findElement(CITY_FIELD).sendKeys(cityName);
+    }
+    
+    public void enterCountry(String countryName) {
+        findElement(COUNTRY_FIELD).clear();
+        findElement(COUNTRY_FIELD).sendKeys(countryName);
+    }
+    
+    public void enterPostalCode(String postalCode) {
+        findElement(POSTAL_CODE_FIELD).clear();
+        findElement(POSTAL_CODE_FIELD).sendKeys(postalCode);
+    }
+    
+    public void selectParentLocation(String locationName) {
+        findElement(PARENT_LOCATION_FIELD).clear();
```

Comment:
Are you sure you are testing parent location? Not postal code?

Reply:
It is Parent Location

![ParentLocation](https://user-images.githubusercontent.com/38831673/132353670-ba68c0ad-39d9-4b0e-b171-d50cb412b8ff.png)

Reply:
Are you sure that this is testing parent location?

Reply:
Based on the drop down default name, it shows Parent Location

![Location](https://user-images.githubusercontent.com/38831673/132366011-98a23236-bfa8-467e-9118-f5569cb952ec.png)

@dkayiwa if not Parent Location then what is  it testing ?

Reply:
@kdaud take a step back and critically look again at this code in detail.

Reply:
@dkayiwa I took the step back and critically looked again at the code in detail. Then  refactored the logic of `selectParentLocation()` to rather use `selectFrom()` than using `findElement().sendKeys()`.